component/src/main/java/io/siddhi/extension/io/file/listeners/FileCronExecutor.java

```diff
@@ -0,0 +1,212 @@
+import org.apache.log4j.Logger;
+import org.quartz.CronScheduleBuilder;
+import org.quartz.Job;
+import org.quartz.JobBuilder;
+import org.quartz.JobDataMap;
+import org.quartz.JobDetail;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.JobKey;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.Trigger;
+import org.quartz.TriggerBuilder;
+import org.quartz.impl.StdSchedulerFactory;
+import org.wso2.carbon.messaging.BinaryCarbonMessage;
+import org.wso2.carbon.messaging.exceptions.ClientConnectorException;
+import org.wso2.transport.file.connector.sender.VFSClientConnector;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+import static io.siddhi.extension.io.file.util.GenerateAppProperties.generateProperties;
+
+/**
+ * FileCron Listener is executted when the cron expression is given. If the current time satisfied by the cron
+ * expression then the file processing will be executed.
+ */
+public class FileCronExecutor implements Job {
+    private static final Logger log = Logger.getLogger(FileCronExecutor.class);
+
+    public FileCronExecutor() {
+    }
+
+    public static void scheduleJob(FileSourceConfiguration fileSourceConfiguration,
+                                   SourceEventListener sourceEventListener) {
+        try {
+            JobKey jobKey = new JobKey(Constants.JOB_NAME, Constants.JOB_GROUP);
+
+            Scheduler scheduler = new StdSchedulerFactory().getScheduler();
+            fileSourceConfiguration.setScheduler(scheduler);
+            if (scheduler.checkExists(jobKey)) {
+                scheduler.deleteJob(jobKey);
+            }
+            scheduler.start();
+            // JobDataMap used to access the object in the job class
+            JobDataMap dataMap = new JobDataMap();
+            dataMap.put(Constants.FILE_SOURCE_CONFIGURATION, fileSourceConfiguration);
+            dataMap.put(Constants.SOURCE_EVENT_LISTENER, sourceEventListener);
+            // Define instances of Jobs
+            JobDetail cron = JobBuilder.newJob(FileCronExecutor.class)
+                    .usingJobData(dataMap)
+                    .withIdentity(jobKey)
+                    .build();
+            //Trigger the job to at given cron expression
+            Trigger trigger = TriggerBuilder
+                    .newTrigger()
+                    .withIdentity(Constants.TRIGGER_NAME, Constants.TRIGGER_GROUP)
+                    .withSchedule(
+                            CronScheduleBuilder.cronSchedule(fileSourceConfiguration.getCronExpression())).build();
+            // Tell quartz to schedule the job using our trigger
+            scheduler.scheduleJob(cron, trigger);
+        } catch (SchedulerException e) {
+            log.error("The error occurs at scheduler start : " + e);
```

Comment:
Is it possible to get the siddhi app name over here? you can access the name from `siddhiAppContext.getName()`
Please add that information for all the exceptions accordingly.

Reply:
fixed in 33a0d4ec677b82f844777943dec4b611303e759d